network/commons/keys.go

```diff
@@ -19,14 +19,24 @@ func ECDSAPrivFromInterface(privkey crypto.PrivKey) (*ecdsa.PrivateKey, error) {
 	secpKey := privkey.(*crypto.Secp256k1PrivateKey)
 	rawKey, err := secpKey.Raw()
 	if err != nil {
-		return nil, errors.Wrap(err, "could mot convert ecdsa.PrivateKey")
+		return nil, errors.Wrap(err, "could not convert ecdsa.PrivateKey")
 	}
 
 	privKey := new(ecdsa.PrivateKey)
 	k := new(big.Int).SetBytes(rawKey)
 	privKey.D = k
 	privKey.Curve = gcrypto.S256() // Temporary hack, so libp2p Secp256k1 is recognized as geth Secp256k1 in disc v5.1.
-	privKey.X, privKey.Y = gcrypto.S256().ScalarBaseMult(rawKey)
+
+	ecdhPrivKey, err := ecdh.P256().NewPrivateKey(rawKey)
```

Comment:
Why are we using elliptic curve diffie hellman for that? Is this the standard way?